### PR TITLE
enable errorlint, errname and contextcheck linters

### PR DESCRIPTION
**Pull request by mmorel-35**

### CODE REVIEW

File: kong/admin_service.go [hunk 1]
```diff
@@ -232,7 +232,7 @@ func (s *AdminService) ListWorkspaces(ctx context.Context,
 	var workspaces []*Workspace
 	_, err = s.client.Do(ctx, req, &workspaces)
 	if err != nil {
-		return nil, fmt.Errorf("error updating admin workspaces: %v", err)
+		return nil, fmt.Errorf("error updating admin workspaces: %w", err)
```
Comment (hbagdi) [inline kong/admin_service.go:235]: No, we don't always want to wrap. Please disable the linter that made you change this code.

File: kong/client.go [hunk 2]
```diff
@@ -198,9 +198,6 @@ func (c *Client) DoRAW(ctx context.Context, req *http.Request) (*http.Response,
 	if req == nil {
 		return nil, fmt.Errorf("request cannot be nil")
 	}
-	if ctx == nil {
-		ctx = defaultCtx
-	}
 	req = req.WithContext(ctx)
```
Comment (hbagdi) [inline kong/client.go:201]: https://pkg.go.dev/net/http#Request.WithContext

> The provided ctx must be non-nil.

**Comment by hbagdi** [comment]
Reviewed partially. Ping @rainest @GGabriele for reviews.

**Review by GGabriele — APPROVED** [review]
LGTM

**Review by hbagdi — APPROVED** [review]
v1.44.0 is out!

**Pull request merged**